azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/core/ManagedIdentityCredentialRetriever.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.tools.auth.core;
+
+import com.azure.identity.ManagedIdentityCredential;
+import com.azure.identity.ManagedIdentityCredentialBuilder;
+import com.microsoft.azure.AzureEnvironment;
+import com.microsoft.azure.tools.auth.exception.LoginFailureException;
+import com.microsoft.azure.tools.auth.model.AuthMethod;
+import com.microsoft.azure.tools.auth.model.AzureCredentialWrapper;
+
+public class ManagedIdentityCredentialRetriever extends AbstractCredentialRetriever {
+
+    public ManagedIdentityCredentialRetriever(AzureEnvironment env) {
+        this.env = env;
+    }
+
+    public AzureCredentialWrapper retrieveInternal() throws LoginFailureException {
+        ManagedIdentityCredential managedIdentityCredential = new ManagedIdentityCredentialBuilder().build();
```

Comment:
Managed Identity also supports setting a client id on it. If the user is using user assigned identity. 
I think this might be missing support for that.

Reply:
I will let user to specify the client id in later PR.